http-transport-binding.md

```diff
@@ -0,0 +1,349 @@
+
+This specification does not further define any of the [Cloud Events][CE] event
+attributes.
+
+Two of the event attributes, `contentType` and `data` are handled specially
+and mapped onto HTTP constructs, all other attributes are transferred as
+metadata without further interpretation.
+
+This mapping is intentionally robust against changes, including the addition
+and removal of event attributes, and also accommodates vendor extensions to the
+event metadata. Any mention of event attributes other than `contentType` and
+`data` is exemplary.
+
+### 2.1. contentType Attribute
+
+The `contentType` attribute is assumed to contain a [RFC2046][RFC2046]
+compliant media-type expression.
+
+### 2.2. data Attribute
+
+The `data` attribute is assumed to contain opaque application data that is
+encoded as declared by the `contentType` attribute.
+
+An application is free to hold the information in any in-memory representation
+of its choosing, but as the value is transposed into HTTP as defined in this
+specification, the assumption is that the `data` attribute value is made
+available as a sequence of bytes.
+
+For instance, if the declared `contentType` is
+`application/json;charset=utf-8`, the expectation is that the `data` attribute
+value is made available as [UTF-8][RFC3629] encoded JSON text to HTTP.
+
+## 3. HTTP Message Mapping
+
+The event binding is identical for both HTTP request and response messages.
+
+The content mode is chosen by the sender of the event, which is either the
+requesting or the responding party. Gestures that might allow solicitation of
+events using a particular mode might be defined by an application, but are not
+defined here.
+
+The receiver of the event can distinguish between the two modes by inspecting
+the `Content-Type` header value. If the value is prefixed with the Cloud Events
+media type `application/cloudevents`, indicating the use of a known [event
+format](#14-event-formats), the receiver uses *structured* mode, otherwise it
+defaults to *binary* mode.
+
+If a receiver detects the Cloud Events media type, but with an event format that
+it cannot handle, for instance `application/cloudevents+avro`, it MAY still
+treat the event as binary and forward it to another party as-is.
+
+### 3.1. Binary Content Mode
+
+The *binary* content mode accommodates any shape of event data, and allows for
+efficient transfer and without transcoding effort.
+
+#### 3.1.1. HTTP Content-Type
+
+For the *binary* mode, the HTTP `Content-Type` value maps directly to the Cloud
+Events `contentType` attribute.
+
+#### 3.1.2. Event Data Encoding
+
+The [`data` attribute](#22-data-attribute) byte-sequence is used as the HTTP
+message body.
+
+#### 3.1.3. Metadata Headers
+
+All [Cloud Events][CE] attributes with exception of `contentType` and `data`
+are individually mapped to and from distinct HTTP message headers.
+
+##### 3.1.3.1 HTTP Header Names
+
+The naming convention for the HTTP header mapping of attributes is:
```

Comment:
Where does the set of "extension" attributes go? I assuming as HTTP headers, but we probably need to prefix them with something like `CE-x-...`

Reply:
Yes, it'll make sense to promote each extension to being its own field. `CE-X-` seems ok. I'l add that.

Reply:
Looks good thanks

Reply:
+1 to `CE-X-`